src/utils/Utils.ts

```diff
@@ -1607,4 +1608,95 @@ export default class Utils {
   public static hash(data: string): string {
     return crypto.createHash('sha256').update(data).digest('hex');
   }
+
+  public static findSiteAreaInTrees(siteAreaId: string, siteAreaTreeList: SiteArea[] = null, siteAreaTreeNode: SiteArea = null): SiteArea {
```

Comment:
I think that siteAreaTreeNode can be removed.
What about something like this (I haven't tested it)

```
  public static findSiteAreaInTrees(siteAreaID: string, siteAreaTrees: SiteArea[] = []): SiteArea {
    if (!Utils.isEmptyArray(siteAreaTrees)) {
      // Loop through trees to find specific tree
      for (const siteAreaTree of siteAreaTrees) {
        // Check node
        if (siteAreaTree.id === siteAreaID) {
          return siteAreaTree;
        }
        // Site Area ID is defined return tree, which contains the site area
        const requestedTree = this.findSiteAreaInTrees(siteAreaID, siteAreaTree.siteAreaChildren);
        if (requestedTree) {
          return siteAreaTree;
        }
      }
    }
  }

```

Reply:
This should work. 👍